src/query/block/lazy.go

```diff
@@ -83,12 +83,20 @@ func (it *lazyStepIter) Meta() Metadata {
 }
 
 func (it *lazyStepIter) Current() Step {
-	c := it.it.Current()
-	tt := it.opts.TimeTransform()
+	var (
+		c        = it.it.Current()
+		tt, vt   = it.opts.TimeTransform(), it.opts.ValueTransform()
+		stepVals = c.Values()
+	)
+
+	vals := make([]float64, len(stepVals))
```

Comment:
nit: could you use the `make([].., 0, ..)` formulation and append instead of directly accessing by idx.

Reply:
Isn't it more efficient to allocate the length up front (instead of capacity) and index in? Or is this a better safe than sorry situation and avoid a potential panic?

Reply:
can get both by doing the other version. 

i.e. when using the version with append (below), it's not expanding the slice unless the initial alloc wasn't large enough. so you still get the benefit of the up front alloc but also the safety of not panic'ing. 
```
slice := make([]type, 0, len)
for _, v := range otherThing {
  slice = append(slice, v)
}
```

Reply:
Ah okay, yeah that makes sense.